soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -15,32 +15,59 @@
 #include "objects/object_gi_fire/object_gi_fire.h"
 #include "objects/object_fish/object_fish.h"
 #include "objects/object_toki_objects/object_toki_objects.h"
+#include "objects/object_gi_bomb_2/object_gi_bomb_2.h"
 #include "objects/gameplay_field_keep/gameplay_field_keep.h"
 #include "soh_assets.h"
+#include "dungeon.h"
 #include "soh/Enhancements/cosmetics/cosmeticsTypes.h"
 
 extern "C" {
 extern SaveContext gSaveContext;
 }
 
+    const char* SmallBaseCvarValue[10] = {
+        CVAR_COSMETIC("Key.ForestSmallBase.Value"),
+        CVAR_COSMETIC("Key.FireSmallBase.Value"),
+        CVAR_COSMETIC("Key.WaterSmallBase.Value"),
+        CVAR_COSMETIC("Key.SpiritSmallBase.Value"),
+        CVAR_COSMETIC("Key.ShadowSmallBase.Value"),
+        CVAR_COSMETIC("Key.WellSmallBase.Value"),
+        CVAR_COSMETIC("Key.GTGSmallBase.Value"),
+        CVAR_COSMETIC("Key.FortSmallBase.Value"),
+        CVAR_COSMETIC("Key.GanonsSmallBase.Value"),
+        CVAR_COSMETIC("Key.ChestSmallBase.Value"),
+    };
+
+    const char* SmallIconCvarValue[10] = {
+        CVAR_COSMETIC("Key.ForestIcon.Value"),
+        CVAR_COSMETIC("Key.FireIcon.Value"),
+        CVAR_COSMETIC("Key.WaterIcon.Value"),
+        CVAR_COSMETIC("Key.SpiritIcon.Value"),
+        CVAR_COSMETIC("Key.ShadowIcon.Value"),
+        CVAR_COSMETIC("Key.WellIcon.Value"),
+        CVAR_COSMETIC("Key.GTGIcon.Value"),
+        CVAR_COSMETIC("Key.FortIcon.Value"),
+        CVAR_COSMETIC("Key.GanonsIcon.Value"),
+        CVAR_COSMETIC("Key.ChestIcon.Value"),
+    };
+
 extern "C" u8 Randomizer_GetSettingValue(RandomizerSettingKey randoSettingKey);
 
 extern "C" void Randomizer_DrawSmallKey(PlayState* play, GetItemEntry* getItemEntry) {
-    s8 keysCanBeOutsideDungeon = getItemEntry->getItemId == RG_GERUDO_FORTRESS_SMALL_KEY ?
-        Randomizer_GetSettingValue(RSK_GERUDO_KEYS) != RO_GERUDO_KEYS_VANILLA :
-        DUNGEON_ITEMS_CAN_BE_OUTSIDE_DUNGEON(RSK_KEYSANITY);
-    s8 isColoredKeysEnabled = keysCanBeOutsideDungeon && CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("MatchKeyColors"), 1);
-    s16 color_slot = getItemEntry->getItemId - RG_FOREST_TEMPLE_SMALL_KEY;
-    s16 colors[9][3] = {
-        { 4, 195, 46 },    // Forest Temple
-        { 237, 95, 95 },   // Fire Temple
-        { 85, 180, 223 },  // Water Temple
-        { 222, 158, 47 },  // Spirit Temple
-        { 126, 16, 177 },  // Shadow Temple
-        { 227, 110, 255 }, // Bottom of the Well
-        { 221, 212, 60 },  // Gerudo Training Grounds
-        { 255, 255, 255 }, // Thieves' Hideout
-        { 80, 80, 80 }     // Ganon's Castle
+    s8 isCustomKeysEnabled = CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("CustomKeyModels"), 0);
+    int slot = getItemEntry->drawItemId - RG_FOREST_TEMPLE_SMALL_KEY;
+
+    Gfx* CustomdLists[] = {
```

Comment:
Probably would be a good idea to rename `CustomdLists` to something like `customIconDLs`.